hub/ui-new/src/components/basic-detail/BasicDetail.tsx

```diff
@@ -32,149 +32,180 @@ import './basicdetail.css';
 import {fetchTaskDescription} from '../redux/Actions/TaskActionDescription';
 import {connect} from 'react-redux';
 import Rating from '../rating/Rating';
+import {API_URL} from '../../constants';
+
+
+export interface LatestVersionInfo {
+  id: number,
+  version: string,
+  displayName: string,
+  description: string,
+  minPipelinesVersion: string,
+  rawURL: string,
+  webURL: string,
+  updatedAt: string,
+}
+export interface CatalogInfo {
+  id: number,
+  type: string,
+}
+export interface TagInfo {
+  id: number,
+  name: string,
+}
 export interface BasicDetailPropObject {
   id: any
   name: string;
-  description: string;
+  kind: string,
+  catalog: CatalogInfo,
+  latestVersion: LatestVersionInfo,
+  tags: Array<TagInfo>,
   rating: number;
-  latestVersion: string,
-  tags: [],
-  type: string,
-  data: [],
-  displayName: string
 }
 
 export interface Version {
+  id: number,
   version: string,
-  description: string,
-  rawUrl: string,
-  webUrl: string,
-  displayName: string
+  rawURL: string,
+  webURL: string,
 }
-
 export interface BasicDetailProp {
   task: BasicDetailPropObject
-  version: Version
+  version: Array<Version>
 }
 
 
 const BasicDetail: React.FC<BasicDetailProp> = (props: any) => {
   React.useEffect(() => {
-    props.fetchTaskDescription(props.version.rawUrl);
+    props.fetchTaskDescription(props.task.latestVersion.rawURL);
     // eslint-disable-next-line
   }, [])
 
+
   const taskArr: any = [];
   const [isModalOpen, setIsModalOpen] = useState(false);
 
-  const [summary, setSummary] = useState(props.task.description.substring(0,
-    props.task.description.indexOf('\n')));
+  const [summary, setSummary] = useState(
+    props.task.latestVersion.description.substring(0,
+      props.task.latestVersion.description.indexOf('\n')) ||
+    props.task.latestVersion.description,
+  );
 
   const [descrption, setDescription] =
     useState(
-      props.task.description.substring(props.task.description.indexOf('\n') +
-        1).trim());
+      props.task.latestVersion.description.indexOf('\n') !== -1 ?
```

Comment:
indexOf("\n") is used at multiple places, perhaps it is better to store and reuse.